src/java.base/share/classes/java/io/ObjectInputStream.java

```diff
@@ -2138,7 +2138,9 @@ private Object readArray(boolean unshared) throws IOException {
 
         ObjectStreamClass desc = readClassDesc(false);
         int len = bin.readInt();
-
+        if (len < 0) {
+            throw new InvalidClassException(desc.getName(), "Array length < 0 (" + len + ")");
```

Comment:
I'd keep the exception logic simple/minimal.  Especially since this is a manual and exceptional case that is not the result of any input provided through an API.
```
throw new InvalidClassException(desc.getName(), "Array length is negative");
```

Reply:
Done.